test/e2e/node_drain.go

```diff
@@ -64,6 +65,16 @@ type NodeDrainTimeoutSpecInput struct {
 	// Allows to inject a function to be run after test namespace is created.
 	// If not specified, this is a no-op.
 	PostNamespaceCreated func(managementClusterProxy framework.ClusterProxy, workloadClusterNamespace string)
+
+	// Enables additional verification for volumes blocking machine deletion.
+	// Requires to add appropriate resources via CreateAdditionalResources.
+	VerifyNodeVolumeDetach bool
+
+	// Allows to overwrite the default function used for unblocking volume detachments.
+	UnblockNodeVolumeDetachment func(ctx context.Context, bootstrapClusterProxy framework.ClusterProxy, cluster *clusterv1.Cluster, waitControlPlaneIntervals, waitWorkerNodeIntervals []interface{})
+
+	// Allows to create additional resources.
+	AdditionalResources func(clusterProxy framework.ClusterProxy, cluster *clusterv1.Cluster, controlPlane *controlplanev1.KubeadmControlPlane, mds []*clusterv1.MachineDeployment)
```

Comment:
```suggestion
	CreateAdditionalResources func(clusterProxy framework.ClusterProxy, cluster *clusterv1.Cluster)
```
We should not make assumptions that everyone using the E2E test framework is using KCP.
Also, considering we are not using controlPlane and mds, might be let's simply drop them (if necessary, they can be retrived inside CreateAdditionalResources starting from the cluster)